### PR TITLE
fix: payload name not nullable

### PR DESCRIPTION
**Pull request by lzap**
https://issues.redhat.com/browse/HMSPROV-373

### CODE REVIEW

File: internal/services/aws_reservation_service.go [hunk 1]
```diff
@@ -48,8 +48,8 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 	reservation.Status = "Created"
 	reservation.Provider = models.ProviderTypeAWS
 	reservation.Steps = 2
-	if payload.Name != nil {
-		newName := config.Application.InstancePrefix + *payload.Name
+	if config.Application.InstancePrefix != "" {
```
Comment (ezr-ondrej) [inline internal/services/aws_reservation_service.go:51]: I belive this is introducing a regression, that the CI is finding.
Now you can't name the instance if the prefix is not configured.

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
The commit message failure is a bug in the script: #324
But the IQE is correct about there being a regression I believe

File: internal/services/aws_reservation_service.go (hunk 1 above)
Comment (lzap) [reply internal/services/aws_reservation_service.go]: Yeah I rushed that in, fixed.

File: internal/services/aws_reservation_service.go [hunk 2]
```diff
@@ -48,10 +48,8 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 	reservation.Status = "Created"
 	reservation.Provider = models.ProviderTypeAWS
 	reservation.Steps = 2
-	if payload.Name != nil {
-		newName := config.Application.InstancePrefix + *payload.Name
-		reservation.Detail.Name = &newName
-	}
+	newName := config.Application.InstancePrefix + payload.Name
+	reservation.Detail.Name = &newName
```
Comment (ezr-ondrej) [inline internal/services/aws_reservation_service.go:52]: Will this behave correctly when there is an empty name?
Before we had nil for name not set, now we have empty string.
I don't know if that matters for AWS, but I'd rather be sure (our IQE pipeline does not test this as it is setting the name).

**Review by akhil-jha — APPROVED** [review]

File: internal/services/aws_reservation_service.go (hunk 2 above)
Comment (lzap) [reply internal/services/aws_reservation_service.go:52]: What you mean? I believe that `"" + "" = ""`.

Comment (ezr-ondrej) [reply internal/services/aws_reservation_service.go:52]: oki make sense :) I was talking about what we send down the road :)

**Pull request merged**